.github/workflows/rust.yml

```diff
@@ -20,3 +20,7 @@ jobs:
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+    - name: Check formatting
+      run: cargo fmt
+    - name: Clippy
+      run: cargo clippy -- -D warnings
```

Comment:
I added `cargo fmt` and `cargo clippy` checks too, I think they are useful to maintain the codebase consistent (and clippy suggestions tend to be pretty good too)

Reply:
Thanks @luizirber !